staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -83,16 +87,31 @@ func NewCRDStructuredMergeManager(models openapiproto.Models, objectConverter ru
 		updater: merge.Updater{
 			Converter: internal.NewCRDVersionConverter(typeConverter, objectConverter, hub),
 		},
+		preparator: preparator,
 	}, nil
 }
 
 // Update implements Manager.
 func (f *structuredMergeManager) Update(liveObj, newObj runtime.Object, managed Managed, manager string) (runtime.Object, Managed, error) {
-	newObjVersioned, err := f.toVersioned(newObj)
+	newObjUnversioned, err := f.toUnversioned(newObj)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to convert new object to unversioned: %v", err)
+	}
+	liveObjUnversioned, err := f.toUnversioned(liveObj)
```

Comment:
@apelisse it seems like i still have to do the conversion roundtrip for both objects as we don't get unversioned.
The reason we do .toVersioned below seems to be ensuring the right version?